src/resonate/resonate.py

```diff
@@ -114,40 +113,79 @@ def register(
             Concatenate[Context, P],
             Generator[Yieldable, Any, T],
         ],
+        /,
+        *,
         name: str | None = None,
         version: int = 1,
-        retry_policy: retry_policy.RetryPolicy | None = None,
-    ) -> RegisteredFn[P, T]: ...
+        retry_policy: RetryPolicy | None = None,
+    ) -> None: ...
     @overload
     def register(
         self,
         func: Callable[Concatenate[Context, P], Coroutine[Any, Any, T]],
+        /,
+        *,
         name: str | None = None,
         version: int = 1,
-        retry_policy: retry_policy.RetryPolicy | None = None,
-    ) -> RegisteredFn[P, T]: ...
+        retry_policy: RetryPolicy | None = None,
+    ) -> None: ...
     @overload
     def register(
         self,
         func: Callable[Concatenate[Context, P], T],
+        /,
+        *,
         name: str | None = None,
         version: int = 1,
-        retry_policy: retry_policy.RetryPolicy | None = None,
-    ) -> RegisteredFn[P, T]: ...
+        retry_policy: RetryPolicy | None = None,
+    ) -> None: ...
+    @overload
     def register(
         self,
-        func: DurableCoro[P, T] | DurableFn[P, T],
+        *,
         name: str | None = None,
         version: int = 1,
-        retry_policy: retry_policy.RetryPolicy | None = None,
-    ) -> RegisteredFn[P, T]:
-        if name is None:
-            name = func.__name__
-        self._registry.add(
-            name,
-            (func, Options(version=version, durable=True, retry_policy=retry_policy)),
-        )
-        return RegisteredFn[P, T](self._scheduler, func)
+        retry_policy: RetryPolicy | None = None,
+    ) -> Callable[
+        [Callable[Concatenate[Context, P], Any]],
+        RegisteredFn[P, Any],
+    ]: ...
+    def register(
+        self, *args: Any, **kwargs: Any
+    ) -> (
+        None
+        | Callable[
+            [Callable[Concatenate[Context, P], Any]],
+            RegisteredFn[P, Any],
+        ]
+    ):
+        name: str | None = kwargs.get("name")
+        version: int = kwargs.get("version", 1)
+        retry_policy: RetryPolicy | None = kwargs.get("retry_policy")
+        if args and callable(args[0]):
+            func: Callable = args[0]  # type: ignore[type-arg]
+            self._registry.add(
+                name or func.__name__,
+                (
+                    func,
+                    Options(version=version, durable=True, retry_policy=retry_policy),
+                ),
+            )
+            return None
```

Comment:
Can we still return a registered function here I can think of two possibilities:
1. maintain backwards compatibility when wrapping with ` @resonate.register`
2. Use the registered function like `rf = resonate.register()`

Reply:
Cool. I'll change that as soon as I get back home